--- pinentry-0.6.9/pinentry/Makefile.am.orig	Thu May  9 14:38:06 2002
+++ pinentry-0.6.9/pinentry/Makefile.am	Thu Aug 21 10:50:04 2003
@@ -27,7 +27,7 @@
 
 noinst_LIBRARIES = libpinentry.a $(pinentry_curses)
 
-AM_CPPFLAGS = -I$(top_srcdir)/assuan -I$(top_srcdir)/secmem
+AM_CPPFLAGS = -I$(top_srcdir)/secmem
 
 libpinentry_a_SOURCES = pinentry.h pinentry.c
 libpinentry_curses_a_SOURCES = pinentry-curses.h pinentry-curses.c
--- pinentry-0.6.9/curses/Makefile.am.orig	Thu May  9 14:38:06 2002
+++ pinentry-0.6.9/curses/Makefile.am	Thu Aug 21 10:49:29 2003
@@ -23,7 +23,7 @@
 
 AM_CPPFLAGS = $(NCURSES_INCLUDE) -I$(top_srcdir)/pinentry
 LDADD = ../pinentry/libpinentry.a ../pinentry/libpinentry-curses.a \
-	../assuan/libassuan.a ../secmem/libsecmem.a \
+	-lassuan ../secmem/libsecmem.a \
 	$(LIBCAP) $(LIBCURSES) $(LIBICONV)
 
 pinentry_curses_SOURCES = pinentry-curses.c
--- pinentry-0.6.9/gtk/Makefile.am.orig	Thu May  9 14:38:06 2002
+++ pinentry-0.6.9/gtk/Makefile.am	Thu Aug 21 10:49:43 2003
@@ -31,7 +31,7 @@
 
 AM_CPPFLAGS = $(GLIB_CFLAGS) $(GTK_CFLAGS) $(ncurses_include) \
 	-I$(top_srcdir)/secmem -I$(top_srcdir)/pinentry
-LDADD = ../pinentry/libpinentry.a ../assuan/libassuan.a ../secmem/libsecmem.a \
+LDADD = ../pinentry/libpinentry.a -lassuan ../secmem/libsecmem.a \
 	$(LIBCAP) $(GTK_LIBS) $(libcurses)
 
 pinentry_gtk_SOURCES = pinentry-gtk.c \
--- pinentry-0.7.0/qt/Makefile.am.orig	2003-12-23 09:39:00.000000000 +0100
+++ pinentry-0.7.0/qt/Makefile.am	2003-12-24 02:54:32.645525240 +0100
@@ -32,10 +32,10 @@
 endif
 
 
-AM_CPPFLAGS = -I$(top_srcdir)/assuan -I$(top_srcdir)/secmem $(QT_INCLUDES) $(ncurses_include)
+AM_CPPFLAGS = -I$(top_srcdir)/secmem $(QT_INCLUDES) $(ncurses_include)
 AM_CXXFLAGS = $(QT_CXXFLAGS)
 pinentry_qt_LDADD = $(QT_LIBS) $(libcurses) ../pinentry/libpinentry.a \
-	$(top_builddir)/assuan/libassuan.a \
+	-lassuan \
 	$(top_builddir)/secmem/libsecmem.a $(LIBCAP)
 pinentry_qt_LDFLAGS = $(QT_LDFLAGS)
 
--- pinentry-0.6.9/Makefile.am.orig	Tue Dec 17 11:14:29 2002
+++ pinentry-0.6.9/Makefile.am	Thu Aug 21 10:49:12 2003
@@ -39,7 +39,7 @@
 pinentry_qt = 
 endif
 
-SUBDIRS = assuan secmem pinentry ${pinentry_curses} \
+SUBDIRS = secmem pinentry ${pinentry_curses} \
 	${pinentry_gtk} ${pinentry_qt} doc
 
 signed-dist: $(distdir).tar.gz.sig
--- pinentry-0.7.0/pinentry/pinentry.c.orig	2003-12-22 19:28:07.000000000 +0100
+++ pinentry-0.7.0/pinentry/pinentry.c	2003-12-24 03:29:35.018915904 +0100
@@ -37,6 +37,7 @@
 #include "secmem-util.h"
 #include "pinentry.h"
 
+#define ASSUAN_Locale_Problem 127
 
 /* Keep the name of our program here. */
 static char this_pgmname[50]; 
@@ -649,8 +650,6 @@
   for (i = j = 0; table[i].name; i++)
     {
       rc = assuan_register_command (ctx,
-                                    table[i].cmd_id ? table[i].cmd_id
-                                                   : (ASSUAN_CMD_USER + j++),
                                     table[i].name, table[i].handler);
       if (rc)
         return rc;
